src/main/java/org/openmainframeproject/tersedecompress/TerseDecompress.java

```diff
@@ -95,18 +103,49 @@ else if (outputFileName == null)
     		printUsageAndExit();
     	}
 
+		System.out.println("Attempting to decompress input file (" + inputFileName +") to output file (" + outputFileName +")");
+
+		if (outputFileName.endsWith(".gz")) {
+			try (TerseDecompresser outputWriter = TerseDecompresser.create(new FileInputStream(inputFileName), new GZIPOutputStream(new FileOutputStream(outputFileName), 8192, true)))
+			{	 
+				outputWriter.TextFlag = textMode;
+				outputWriter.decode();
+			}
+		}
+		else {
+			try (TerseDecompresser outputWriter = TerseDecompresser.create(new FileInputStream(inputFileName), new FileOutputStream(outputFileName)))
+			{	 
+				outputWriter.TextFlag = textMode;
+				outputWriter.decode();
+			}
+		}
 
-        try (TerseDecompresser outputWriter 
-        		= TerseDecompresser.create(new FileInputStream(inputFileName), new FileOutputStream(outputFileName)))
-        {	 
-        	outputWriter.TextFlag = textMode;
-	        System.out.println("Attempting to decompress input file (" + inputFileName +") to output file (" + outputFileName +")");
-	        outputWriter.decode();
-        }	
-		
         System.out.println("Processing completed");
     }
 
+    private void parseArgs(String args[]) {
+        for (int i = 0; i < args.length; i++) {
+            if (args[i].equals("-h") || args[i].equals("--help")) {
+                isHelpRequested = true;
+            }
+            else if (args[i].equals("-b")) {
+                textMode = false;
+            }
+            // first non-flag argument is the input file name
+            else if (inputFileName == null) {
+                inputFileName = args[i];
+            }
+            // second non-flag argument is the input file name
```

Comment:
looks like typo, should it be "second non-flag argument is the output file name" ?